h2/src/main/org/h2/index/IndexCondition.java

```diff
@@ -155,7 +155,7 @@ public Value getCurrentValue(Session session) {
             v = column.convert(v);
             valueSet.add(v);
         }
-        Value[] array = valueSet.toArray(new Value[valueSet.size()]);
+        Value[] array = valueSet.toArray(new Value[0]);
```

Comment:
Please, revert it. `ArrayList.toArray(new T[0])` works faster than `ArrayList.toArray(new T[list.size()]`), but other collections are usually not. I checked all collections some time ago with JMH tests with different Java versions and decided to use 0 only for `ArrayList` based on results of these tests.

Reply:
That's interesting - I'll revert it. 

BTW do you have those JMHs somewhere on github?

Reply:
No, I had a lot of such tests, but I didn't publish code anywhere and I probably already overwrote or deleted them, I can't find them now.

You can check this article, it has some samples and a lot of results:
https://shipilev.net/blog/2016/arrays-wisdom-ancients/

But notice that you actually perform this change in a more complex collections, like key set views of maps, they are not implemented in the same way as `ArrayList.toArray(T[])`. They are not tested in that article, so additional testing is required.

Or you may just leave this as is.